bin/ocFunctions.inc

```diff
@@ -1799,6 +1799,75 @@ function getSecret() {
   )
 }
 
+function buildPromptCreds(){
+  namespaceName=${1}
+  newCredName=${2}
+  dockerReg=${3}
+  credEnvs=${4}
+  userName=${5}
+  password=${6}
+  
+  if ([ -z ${userName} ] || [ -z ${password} ]); then
+    echoWarning "Please enter your ${dockerReg} username"
+    read userName
+
+    echoWarning "Please enter your ${dockerReg} password"
+    read password
+  fi
+
+  #set cred env to tools if unset
+  if [ -z "${credEnvs}" ]; then
+    credEnvs="tools"
+  fi
+  #create pull secret in each environment we need
+  for env in ${credEnvs}; do
+    cred_exists=0
+    
+    oc create secret docker-registry ${newCredName} -n ${namespaceName}-${env} \
+      --docker-server=${dockerReg} \
+      --docker-username=${userName} \
+      --docker-password=${password} \
+      --docker-email=unused &> /dev/null || cred_exists=1
+    if (( ! ${cred_exists})); then
+      oc secrets link default ${newCredName} --for=pull
```

Comment:
`--for=pull` is not required, see [docs](https://developer.gov.bc.ca/Artifact-Repositories)

Reply:
--for=pull removed